settings.gradle.kts

```diff
@@ -329,8 +329,8 @@ gradle.settingsEvaluated {
         return@settingsEvaluated
     }
 
-    if (!JavaVersion.current().isJava11) {
-        throw GradleException("This build requires JDK 11. It's currently ${getBuildJavaHome()}. You can ignore this check by passing '-Dorg.gradle.ignoreBuildJavaVersionCheck=true'.")
+    if (JavaVersion.current() != JavaVersion.VERSION_17) {
```

Comment:
❓ is there a good reason we don't additionally use the Daemon Toolchain support to avoid tripping this every time we switch between prior and current versions?

https://docs.gradle.org/8.10.2/userguide/gradle_daemon.html#sec:daemon_jvm_criteria

Reply:
No good reason, just don’ didn’t feel like tinkering more than I had to in this pr.

It would probably be good if we dogfooded this feature

Reply:
@ljacomet or @big-guy do you think we should use daemon toolchains here?

Reply:
We were planning to dogfood this feature, I just hadn't gotten around to it yet. It's on my TODO list.

Reply:
https://github.com/gradle/gradle/issues/30773